Odo IBM Z updates

SNiemann15 opened the pull request:
These are updates to odo for IBM Z GA Sep 22

SNiemann15 commented:
@boczkowska Would you please peer review my updates? 
Xiao (Megan) Mei Zheng has already reviewed the content. 

Docs apply to enterprise-4.5 and future releases.

boczkowska commented:
LGTM

Preeticp commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -40,46 +40,51 @@ In `{odo-title}`, services are provisioned from the OpenShift Service Catalog an
 === Officially supported languages and corresponding container images
 
-.Supported languages, container images, and package managers
+.Supported languages, container images, package managers, and supported platforms
```
supported platforms>remove supported as it is mentioned right at the start.

Preeticp commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -40,46 +40,51 @@ In `{odo-title}`, services are provisioned from the OpenShift Service Catalog an
 [options="header"]
 |===
-|Language |Container image |Package manager
+|Language |Container image |Package manager |Supported platform
```
Remove 'Supported'

Preeticp commented:
/lgtm

bergerhoffer commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -40,46 +40,51 @@ In `{odo-title}`, services are provisioned from the OpenShift Service Catalog an
+|amd64, s390x
 
 |
 |https://catalog.redhat.com/software/containers/rhscl/nodejs-12-rhel7/[rhscl/nodejs-12-rhel7]
```
I know you didn't change this link, but it doesn't work, if that's something you could look into

SNiemann15 replied:
I'll ask someone in the odo-team about the links.

bergerhoffer commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -40,46 +40,51 @@ In `{odo-title}`, services are provisioned from the OpenShift Service Catalog an
 
 |
-|https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/nodejs-8-rhel7[rhscl/nodejs-8-rhel7]
+|https://registry.centos.org/centos/nodejs-10-centos7/latest[centos/nodejs-10-centos7]
```
This link doesn't work either, should it?

SNiemann15 replied:
Sorry I didn't think about validating the existing links. I got two now that work and will replace.

bergerhoffer replied:
Thank you!

PanYaLian replied:
Will you use "https://hub.docker.com/r/centos/nodejs-10-centos7" to replace  "https://registry.centos.org/centos/nodejs-10-centos7/latest"

zhengxiaomei123 replied:
For nodejs-10-centos7 , the upstream doc uses this link: https://github.com/sclorg/s2i-nodejs-container
For nodejs-12-centos7 , the upstream doc uses this link: https://github.com/sclorg/s2i-nodejs-container
@SNiemann15 Hope it will help you.

SNiemann15 replied:
@boczkowska said she will update the urls. Yana will you or shall I give it another try and see if my "git" problems still persist?

PanYaLian commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -40,46 +40,51 @@ In `{odo-title}`, services are provisioned from the OpenShift Service Catalog an
+|amd64, s390x
 
 |
 |https://access.redhat.com/containers/#/registry.access.redhat.com/openjdk/openjdk-11-rhel8[openjdk/openjdk-11-rhel8]
```
It looks that https://access.redhat.com/containers/#/registry.access.redhat.com/openjdk/openjdk-11-rhel8 is now Deprecated, will you use the new one "https://access.redhat.com/containers/#/registry.access.redhat.com/ubi8/openjdk-11/"

SNiemann15 replied:
@boczkowska Same here.

boczkowska replied:
odo does not support `ubi8` image currently.  For questions and discussions about it, file an issue at https://github.com/openshift/odo/pulls or join the openshiftdo.slack.com

boczkowska replied:
>Will you use "https://hub.docker.com/r/centos/nodejs-10-centos7" to replace "https://registry.centos.org/centos/nodejs-10-centos7/latest"

Not that I'm aware of.

Preeticp commented:
@bergerhoffer, @boczkowska will create another PR to fix links on existing docs after verifying with the odo devs and QE. We probably don't need to block this PR on account of those links. CC: @SNiemann15 @vikram-redhat

vikram-redhat commented:
> @bergerhoffer, @boczkowska will create another PR to fix links on existing docs after verifying with the odo devs and QE. We probably don't need to block this PR on account of those links. CC: @SNiemann15 @vikram-redhat

If that PR comes immediately after, then that is ok. Otherwise we will be merging links to upstream images in product docs which we can't allow without @sferich888 or @RickJWagner acking.

In any case, the downstream links for the product are available now, so it should just be a matter of @SNiemann15 updating this PR to use the correct links and we can merge.

boczkowska commented:
@bergerhoffer @vikram-redhat I have the confirmed by the QE list of supported images for downstream odo build and sent a PR agains @SNiemann15 's PR here https://github.com/SNiemann15/openshift-docs/pull/1 (Basically, only three URLs need to be updated, everything else is correct).
Once it is merged, it can all be merged together in this PR. :smile:

bergerhoffer commented on cli_reference/developer_cli_odo/understanding-odo.adoc:
```diff
@@ -40,46 +40,51 @@ In `{odo-title}`, services are provisioned from the OpenShift Service Catalog an
 
 |
-|https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/nodejs-8-rhel7[rhscl/nodejs-8-rhel7]
+|https://github.com/sclorg/s2i-nodejs-container[centos/nodejs-10-centos7]
```
@SNiemann15 @boczkowska Is there a reason that these are going from a red hat registry image to now pointing to github links? That doesn't seem right to me - aren't there downstream images for these?

boczkowska replied:
These are the images that odo QE provided me with, cc @amitkrout

boczkowska replied:
It is downstream-ready from what I understand
`NodeJS images based on Red Hat Software Collections and intended for OpenShift and general usage, that provide a platform for building and running NodeJS applications. Users can choose between Red Hat Enterprise Linux, Fedora, and CentOS based images. `

vikram-redhat replied:
@boczkowska the correct RH downstream link to that should be: https://access.redhat.com/containers/#/registry.access.redhat.com/rhscl/nodejs-10-rhel7.

The GH repo link is technically not supported and we should replace it with the Red Hat registry image one. I am not sure why QE says that is the right image for downstream (the link is to CentOS image, which is **_definitely_** not correct and not supported by RH).

@sferich888 if you have thoughts on this, please feel free to add.

SNiemann15 commented:
@bergerhoffer I updated the table and removed those images that are not officially supported by RH. @boczkowska verified that with Thomas Kral. 
I went through the squash etc. together with Yana and for some reason that we don't understand 
```
post_installation_configuration/modules
post_installation_configuration/images
```
got into the commit. I haven't touched anything there. So I'm not sure if everything went right here. 

If this is not a problem the updates can be published on Sep 22 as previously planned not today correct @rkratky?

bergerhoffer commented:
@SNiemann15 it looks like when you squashed, you accidentally squashed including a commit that wasn't yours. If you look at your commits tab, it now shows that @vikram-redhat authored it, and you added to it.

Those changes you're seeing are from vikram's commit here (https://github.com/openshift/openshift-docs/commit/8c34735dc4e7ee2b321ffa125fbd41e2c30497f5), which was probably the commit right before your initial commit. And you accidentally squashed all the way to that one.

There might be other ways to recover from this, but below are some steps that should work. Reach out to me on slack if you have any questions on it or want to do it together over bluejeans.

First, let's get rid of those two extra changes you're seeing:

1. Fetch the latest changes

```
$ git fetch upstream
```

2. Rebase your branch with the latest changes. Since the modules and images links exist in the latest changes, this should get rid of them in your diffs once you rebase.

```
$ git rebase upstream/master
```



Now, at least those changes should be gone from your commit. However, your commit is still wrong, showing Vikram as the author. So let's redo the commit. 

1. First, let's find the commit right before yours (technically listed as Vikram's). 

```
$ git log
commit aea808d766d6a552d06367ad1dc49d3f71a0b932 (HEAD -> pr-25138, origin/test-pr)
Author: Vikram Goyal <vigoyal@redhat.com>
Date:   Sat Aug 29 09:15:49 2020 +1000

    odo installation instructions for IBM Z updated urls
    
    removed images not officially supported by RH from table

commit 99afacd5331fa839d9b7cebc72b383996751f097 (upstream/master, test-master)
Merge: dbdb1a1e1 bc9af5f00
Author: Andrea Hoffer <bergerhoffer@users.noreply.github.com>
Date:   Tue Sep 15 11:38:16 2020 -0400

    Merge pull request #25385 from mgarrellRH/CNV4333
    
    CNV - JIRA story CNV-4333 Adding fslist, guestosinfo, and userlist commands to ands to Virtctl commands table in Virtual client commands subsection of Using the CLI tools section

...
```

This is my output, but by the time you do this, your commit history might be different. So I'm choosing the commit ID right below yours, which in my example is `99afacd5331fa839d9b7cebc72b383996751f097`.

2. Do a **SOFT** reset to the commit before yours (technically listed as Vikram's):

```
$ git reset --soft 99afacd5331fa839d9b7cebc72b383996751f097
```

3. Make a new commit:

```
$ git commit -m "odo installation instructions for IBM Z updated urls"
```

4. Push the changes to your branch/PR:

```
$ git push -f origin HEAD
```

SNiemann15 commented:
This is ready to be merged but not before Sep 22nd. @rkratky can you reach out to docs team on that date to get it published thanks!

bergerhoffer commented:
After we resolved the commit issue this morning, these changes LGTM! Ready to be merged once we get the OK from @rkratky on Sept 22.

Preeticp commented:
@bergerhoffer we have confirmation from Robert, this PR and the one for IBM Power: https://github.com/openshift/openshift-docs/pull/25505 need to be merged on Sept 22.

bergerhoffer commented:
Great, just remind us on the 22nd then please!

vikram-redhat commented:
ack from Preeti. Merging and CPing.

vikram-redhat commented:
/cherrypick enterprise-4.6

vikram-redhat commented:
/cherrypick enterprise-4.5
